sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -598,6 +601,46 @@ class FilesElement extends Polymer.Element {
     }
   }
 
+  _toggleAltAddToolbar() {
+    this.$.altAddToolbar.toggle();
+  }
+
+  _toggleAltUpdateToolbar() {
+    this.$.altUpdateToolbar.toggle();
+  }
+
+  /**
+   * Called on window.resize, collapses elements to keep the element usable
+   * on small screens.
+   */
+  _resizeHandler() {
+    const width = this.$.toolbar.clientWidth;
+    // Collapse the add buttons on the toolbar
+    if (width < this._addToolbarCollapseThreshold) {
+      Utils.moveElementChildren(this.$.addToolbar, this.$.altAddToolbar);
+      this.$.altAddToolbarToggle.style.display = "inline-flex";
+    } else {
+      Utils.moveElementChildren(this.$.altAddToolbar, this.$.addToolbar);
+      this.$.altAddToolbarToggle.style.display = "none";
+      this.$.altAddToolbar.close();
+    }
+
+    // Collapse the file operations buttons on the toolbar
+    if (width < this._updateToolbarCollapseThreshold) {
```

Comment:
I'm not very comfortable writing very specialized functions, such as what's needed to fix this. The function will be unreadable and won't be a lot of sense as a standalone routine. I'd rather repeat the code if it's more readable.
Thoughts?